src/chatd.cpp

```diff
@@ -519,14 +525,13 @@ void Connection::reset() //immediate disconnect
 
 bool Connection::sendBuf(Buffer&& buf)
 {
-    if (!isOnline())
+    if (!isLoggedIn() && !isConnected())
         return false;
 //WARNING: ws_send_msg_ex() is destructive to the buffer - it applies the websocket mask directly
 //Copy the data to preserve the original
     auto rc = ws_send_msg_ex(mWebSocket, buf.buf(), buf.dataSize(), 1);
     buf.free(); //just in case, as it's content is xor-ed with the websock datamask so it's unusable
-    bool result = (!rc && isOnline());
-    return result;
+    return rc;
```

Comment:
This is not correct. `ws_send_msg_ex()` returns 0 on success and -1 or some other value on failure. The original code returns success if `rc` is 0 and the connection state didn't go to offline during the send. You should leave it like that.